libdnf/rpm/repo.cpp

```diff
@@ -193,6 +193,33 @@ inline static void result_get_info(LrResult * result, LrResultInfoOption option,
     }
 }
 
+
+// Map string from config option proxy_auth_method to librepo LrAuth value
+static constexpr struct {
+    const char * name;
+    LrAuth code;
+} PROXYAUTHMETHODS[] = {{"none", LR_AUTH_NONE},
+                        {"basic", LR_AUTH_BASIC},
+                        {"digest", LR_AUTH_DIGEST},
+                        {"negotiate", LR_AUTH_NEGOTIATE},
+                        {"ntlm", LR_AUTH_NTLM},
+                        {"digest_ie", LR_AUTH_DIGEST_IE},
+                        {"ntlm_wb", LR_AUTH_NTLM_WB},
+                        {"any", LR_AUTH_ANY}};
+
+static LrAuth string_to_proxy_auth_methods(const std::string & proxy_auth_method_str) noexcept
```

Comment:
If a single value is returned, could you replace `methods` with `method` in all the places?

Reply:
Ok, clarified details over chat; keeping the plural, will extend it with parsing multiple values into a bitmap in the future.